### PR TITLE
[drape] Fixed harfbuzz include on Mac

### PR DESCRIPTION
**Pull request by biodranik**

### CODE REVIEW

**Review by vng — APPROVED** [review]

File: drape/drape_tests/glyph_mng_tests.cpp [hunk 1]
```diff
@@ -11,16 +11,25 @@
...
 #include FT_MODULE_H
-#include <harfbuzz/hb-ft.h>
+// A system dependency is used on these platforms. TODO(AB): There should be a way to build everything from sources.
+#if defined(OMIM_OS_LINUX) || defined(OMIM_OS_WINDOWS)
 #include <harfbuzz/hb.h>
+#include <harfbuzz/hb-ft.h>
+#else
+#include <hb.h>
+#include <hb-ft.h>
+#endif
```
Comment (Osyotr) [inline drape/drape_tests/glyph_mng_tests.cpp:32]:
```suggestion
#include <hb.h>
#include <hb-ft.h>
```
`.../include/harfbuzz` is in `harfbuzz::harfbuzz`'s `INTERFACE_INCLUDE_DIRECTORIES`.

Comment (biodranik) [reply drape/drape_tests/glyph_mng_tests.cpp:32]: @Osyotr can you please clarify your comment? Did you test it on Linux?

Comment (Osyotr) [reply drape/drape_tests/glyph_mng_tests.cpp:32]: `harfbuzz-config.cmake` from ubuntu 24.04:
```cmake
set_target_properties(harfbuzz::harfbuzz PROPERTIES
  INTERFACE_INCLUDE_DIRECTORIES "${PACKAGE_PREFIX_DIR}/include/harfbuzz"
  ...
)
```

So if `#include <hb.h>` doesn't work for you then you're not linking against `harfbuzz::harfbuzz`: https://github.com/organicmaps/organicmaps/blob/088bef549e4e731304f2c48681141f1485b5cb84/drape/CMakeLists.txt#L177

**Comment by Osyotr** [comment]
You also need to add `add_library(harfbuzz::harfbuzz ALIAS harfbuzz)` there to allow harfbuzz to be consumed in a uniform manner:
https://github.com/organicmaps/organicmaps/blob/088bef549e4e731304f2c48681141f1485b5cb84/3party/harfbuzz/CMakeLists.txt#L7
And this should be `harfbuzz::harfbuzz`:
https://github.com/organicmaps/organicmaps/blob/088bef549e4e731304f2c48681141f1485b5cb84/drape/CMakeLists.txt#L177

File: drape/drape_tests/glyph_mng_tests.cpp (hunk 1 above)
Comment (biodranik) [reply drape/drape_tests/glyph_mng_tests.cpp]: @Osyotr Now it still doesn't work.

Comment (biodranik) [reply drape/drape_tests/glyph_mng_tests.cpp]: Ubuntu 22 has this:
```
$ cat /usr/lib/x86_64-linux-gnu/cmake/harfbuzz/harfbuzz-config.cmake | grep INTERFACE_INCLUDE_DIRECTORIES
  INTERFACE_INCLUDE_DIRECTORIES "${_harfbuzz_prefix}/${_harfbuzz_includedir}/harfbuzz"
  INTERFACE_INCLUDE_DIRECTORIES "${_harfbuzz_prefix}/${_harfbuzz_includedir}/harfbuzz"
  INTERFACE_INCLUDE_DIRECTORIES "${_harfbuzz_prefix}/${_harfbuzz_includedir}/harfbuzz"
    INTERFACE_INCLUDE_DIRECTORIES "${_harfbuzz_prefix}/${_harfbuzz_includedir}/harfbuzz"
```

Comment (biodranik) [reply drape/drape_tests/glyph_mng_tests.cpp]:
```
$ ls -a /usr/include/harfbuzz/
.                hb-blob.h        hb-draw.h  hb-glib.h             hb-graphite2.h  hb-ot-deprecated.h  hb-ot-meta.h     hb-ot-var.h      hb-shape.h    hb.h
..               hb-buffer.h      hb-face.h  hb-gobject-enums.h    hb-icu.h        hb-ot-font.h        hb-ot-metrics.h  hb-ot.h          hb-style.h
hb-aat-layout.h  hb-common.h      hb-font.h  hb-gobject-structs.h  hb-map.h        hb-ot-layout.h      hb-ot-name.h     hb-set.h         hb-unicode.h
hb-aat.h         hb-deprecated.h  hb-ft.h    hb-gobject.h          hb-ot-color.h   hb-ot-math.h        hb-ot-shape.h    hb-shape-plan.h  hb-version.h
```

Comment (biodranik) [reply drape/drape_tests/glyph_mng_tests.cpp]: Sorry, my fault, accidentally removed `::harfbuzz` suffix. Thanks for the help!

**Pull request merged**